src/components/NavMenu.tsx

```diff
@@ -0,0 +1,312 @@
+
+	const width = Dimensions.get('window').width * widthPercent;
+	const openOffset = -width;
+	const closeOffset = -2 * width;
+
+	const translateX = useRef(new Animated.Value(-width));
+	const opacity = useRef(new Animated.Value(0));
+
+	const animateSidemenu = useCallback(
+		(open: boolean) => {
+			Animated.timing(translateX.current, {
+				toValue: open ? openOffset : closeOffset,
+				duration: 120,
+				easing: Easing.ease,
+				useNativeDriver: true,
+			}).start();
+
+			Animated.timing(opacity.current, {
+				toValue: open ? 1 : 0,
+				duration: 120,
+				easing: Easing.ease,
+				useNativeDriver: true,
+			}).start();
+		},
+		[openOffset, closeOffset]
+	);
+
+	useEffect(() => {
+		animateSidemenu(openState);
+	}, [openState, animateSidemenu]);
+
+	const panResponder = useRef(
+		PanResponder.create({
+			onMoveShouldSetPanResponder: () => true,
+			onPanResponderMove: (_, gesture) => {
+				if (gesture.dx < width) {
+					translateX.current.setValue(openOffset + gesture.dx);
+				}
+			},
+			onPanResponderRelease: (_, gesture) => {
+				const collapse = gesture.vx < 0; // swiping left
+				onOpenChange(!collapse);
+				if (!collapse) {
+					animateSidemenu(true);
+				}
+			},
+		})
+	).current;
+
+	const overlay = (
+		<Animated.View
+			pointerEvents={openState ? 'auto' : 'none'}
+			style={{
+				position: 'absolute',
+				zIndex: 1,
+				opacity: opacity.current,
+			}}
+		>
+			<TouchableOpacity
+				style={styles.overlay}
+				onPress={() => onOpenChange(!openState)}
+				activeOpacity={1}
+			/>
+		</Animated.View>
+	);
+
+	const linkList = (
+		<FlatList
+			style={styles.menuList}
+			data={links}
+			renderItem={({ item }) => (
+				<TouchableOpacity
+					key={item.route}
+					style={{
+						flexDirection: 'row',
+						alignItems: 'center',
+					}}
+				>
+					<Icon
+						containerStyle={{ width: 50, paddingLeft: 20 }}
+						name={item.iconName}
+						type="ionicon"
+					/>
+					<Text style={styles.link}>{item.text}</Text>
+				</TouchableOpacity>
+			)}
+			keyExtractor={(link) => link.route}
+			horizontal={false}
+			numColumns={1}
+		/>
+	);
+
+	return (
+		<>
+			{overlay}
+			<Animated.View
+				style={{
+					...styles.menu,
+					transform: [{ translateX: translateX.current }],
+				}}
+				{...panResponder.panHandlers}
+			>
+				<Icon
+					containerStyle={{
+						position: 'absolute',
+						top: '50%',
+						right: -10,
+						zIndex: 1,
+						transform: [{ rotate: '90deg' }],
+					}}
+					name="minus"
+					type="feather"
+					size={40}
+					color="#aaa"
+				/>
+				<MenuHeader {...header} />
+				{linkList}
+				<Toggle
+					state={toggleState}
+					onChange={(val: boolean) => onToggleChange(val)}
+				/>
+			</Animated.View>
+		</>
+	);
+};
+
+interface NavProps {
+	menu: Omit<MenuProps, 'widthPercent' | 'openState' | 'onOpenChange'>;
+	title: string;
+	imgSrc?: ImageSourcePropType;
+}
+
+const SIDEBAR_WIDTH_PERCENT = 0.78;
+
+const NavMenu = ({ menu, title, imgSrc }: NavProps) => {
```

Comment:
Nitpick but could we call this "NavBar" or "TopBar" to indicate that component is the thing you see at the top of our app?